src/compiler/gcc.rs

```diff
@@ -530,8 +566,11 @@ pub fn generate_compile_commands(
             "-o".into(),
             path_transformer.to_dist(out_file)?,
         ];
-        // We could do preprocessor_args here, but skip for consistency with msvc
         arguments.extend(dist::osstrings_to_strings(&parsed_args.common_args)?);
+        // We need to include the preprocessor_args here, since the preprocessor run is
+        // configured to only resolve includes, so the preprocessor needs to run on the
+        // dist side again.
+        arguments.extend(dist::osstrings_to_strings(&parsed_args.preprocessor_args)?);
```

Comment:
As per the thread above, I think this is not right. This passes too many flags to the dist builder, and fails when building Gecko. I posted a fix for this in the thread above, mainly:

```diff
diff --git a/src/compiler/gcc.rs b/src/compiler/gcc.rs
index 8ef663b..e7a1354 100644
--- a/src/compiler/gcc.rs
+++ b/src/compiler/gcc.rs
@@ -121,7 +121,7 @@ counted_array!(pub static ARGS: [ArgInfo<ArgData>; _] = [
     take_arg!("--sysroot", PathBuf, Separated, PassThroughPath),
     take_arg!("-A", OsString, Separated, PassThrough),
     take_arg!("-B", PathBuf, CanBeSeparated, PassThroughPath),
-    take_arg!("-D", OsString, CanBeSeparated, PreprocessorArgument),
+    take_arg!("-D", OsString, CanBeSeparated, PassThrough),
     flag!("-E", TooHardFlag),
     take_arg!("-F", PathBuf, CanBeSeparated, PreprocessorArgumentPath),
     take_arg!("-G", OsString, Separated, PassThrough),
@@ -136,7 +136,7 @@ counted_array!(pub static ARGS: [ArgInfo<ArgData>; _] = [
     take_arg!("-MQ", OsString, Separated, PreprocessorArgument),
     take_arg!("-MT", OsString, Separated, DepTarget),
     flag!("-P", TooHardFlag),
-    take_arg!("-U", OsString, CanBeSeparated, PreprocessorArgument),
+    take_arg!("-U", OsString, CanBeSeparated, PassThrough),
     take_arg!("-V", OsString, Separated, PassThrough),
     take_arg!("-Xassembler", OsString, Separated, PassThrough),
     take_arg!("-Xlinker", OsString, Separated, PassThrough),
@@ -549,10 +549,6 @@ pub fn generate_compile_commands(
             path_transformer.to_dist(out_file)?,
         ];
         arguments.extend(dist::osstrings_to_strings(&parsed_args.common_args)?);
-        // We need to include the preprocessor_args here, since the preprocessor run is
-        // configured to only resolve includes, so the preprocessor needs to run on the
-        // dist side again.
-        arguments.extend(dist::osstrings_to_strings(&parsed_args.preprocessor_args)?);
         Some(dist::CompileCommand {
             executable: path_transformer.to_dist(&executable)?,
             arguments: arguments,
```

Is this intentionally not addressed? If so, why?

Reply:
Not intentionally. Sorry, I think I applied these changes, but somehow I lost them accidentally during rebasing.